src/app/modules/shared_models/member_model.nim

```diff
@@ -232,36 +267,63 @@ QtObject:
       isUntrustworthy: bool,
       ) =
     let ind = self.findIndexForMember(pubKey)
-    if(ind == -1):
+    if ind == -1:
       return
 
-    self.items[ind].displayName = displayName
-    self.items[ind].ensName = ensName
-    self.items[ind].isEnsVerified = isEnsVerified
-    self.items[ind].localNickname = localNickname
-    self.items[ind].alias = alias
-    self.items[ind].icon = icon
-    self.items[ind].isContact = isContact
-    self.items[ind].isVerified = isVerified
-    self.items[ind].memberRole = memberRole
-    self.items[ind].joined = joined
-    self.items[ind].isUntrustworthy = isUntrustworthy
+    var roles: seq[int] = @[]
+
+    if self.items[ind].displayName != displayName:
+      self.items[ind].displayName = displayName
+      roles.add(ModelRole.DisplayName.int)
+
+    if self.items[ind].ensName != ensName:
+      self.items[ind].ensName = ensName
+      roles.add(ModelRole.EnsName.int)
+
+    if self.items[ind].localNickname != localNickname:
+      self.items[ind].localNickname = localNickname
+      roles.add(ModelRole.LocalNickname.int)
+
+    if self.items[ind].isEnsVerified != isEnsVerified:
+      self.items[ind].isEnsVerified = isEnsVerified
+      roles.add(ModelRole.IsEnsVerified.int)
+
+    if self.items[ind].alias != alias:
+      self.items[ind].alias = alias
+      roles.add(ModelRole.Alias.int)
+
+    if self.items[ind].icon != icon:
+      self.items[ind].icon = icon
+      roles.add(ModelRole.Icon.int)
+
+    if self.items[ind].isContact != isContact:
+      self.items[ind].isContact = isContact
+      roles.add(ModelRole.IsContact.int)
+
+    if self.items[ind].isVerified != isVerified:
+      self.items[ind].isVerified = isVerified
+      roles.add(ModelRole.IsVerified.int)
+
+    if self.items[ind].memberRole != memberRole:
+      self.items[ind].memberRole = memberRole
+      roles.add(ModelRole.MemberRole.int)
+
+    if self.items[ind].joined != joined:
+      self.items[ind].joined = joined
+      roles.add(ModelRole.Joined.int)
+
+    if self.items[ind].isUntrustworthy != isUntrustworthy:
+      self.items[ind].isUntrustworthy = isUntrustworthy
+      roles.add(ModelRole.IsUntrustworthy.int)
+
+    if roles.len == 0:
+      return
+
+    roles.add(ModelRole.PreferredDisplayName.int)
```

Comment:
I think it would be beneficial to check if any name used for computing preferred display name changed and only then included that role. It's quite important as that role is used for sorting and notifying change on it will trigger sorting.